src/legacy/server/saved_objects/migrations/kibana/kibana_migrator.ts

```diff
@@ -74,9 +74,6 @@ export class KibanaMigrator {
   public awaitMigration = once(async () => {
     const { server } = this.kbnServer;
 
-    // Wait until the plugins have been found an initialized...
-    await this.kbnServer.ready();
```

Comment:
Since the plugin checks the repository which awaits migrations this was causing the system to loop infinitely during startup. Putting this expectation that `all` the plugins should have loaded is unnecessary since the saved object service and any service should expect the elasticsearch service to be available, I think is quite reasonable. Perhaps ES is not available or in a running state but the code is purely dependent on there being an ES service which it should totally be able to access.

Going forward this issue should be completely resolvable on the `new platform` where the dependency should be explicitly defined.

Reply:
I did not know that the `kbnServer.ready` method was only used in the kibana migrator. But if it can be removed I don't see a reason not to do so. There is another way to run code after all plugins have loaded and thats with adding a `afterPluginsInit` listener.